File: stdin_to_stderr.zig
// Copyright (c) Meta Platforms, Inc. and affiliates.
//
// This source code is licensed under both the MIT license found in the
// LICENSE-MIT file in the root directory of this source tree and the Apache
// License, Version 2.0 found in the LICENSE-APACHE file in the root directory
// of this source tree.

const std = @import("std");

pub fn main() !void {
    var general_purpose_allocator = std.heap.GeneralPurposeAllocator(.{}){};
    const gpa = general_purpose_allocator.allocator();

    const in = std.io.getStdIn().reader();
    const contents = try in.readAllAlloc(gpa, std.math.maxInt(usize));

    const out = std.io.getStdErr().writer();
    try out.writeAll(contents);
}

File: stdin_to_stderr.rs
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under both the MIT license found in the
 * LICENSE-MIT file in the root directory of this source tree and the Apache
 * License, Version 2.0 found in the LICENSE-APACHE file in the root directory
 * of this source tree.
 */

//! Read stdin and send it to stderr.

fn main() {
    std::io::copy(&mut std::io::stdin().lock(), &mut std::io::stderr().lock())
        .expect("read stdin and write to stderr");
}
